tests/kafkatest/tests/core/security_test.py

```diff
@@ -82,7 +82,19 @@ def test_client_ssl_endpoint_validation_failure(self, security_protocol, interbr
 
         # now set the certs to have invalid hostnames so we can run the actual test
         SecurityConfig.ssl_stores.valid_hostname = False
-        self.kafka.restart_cluster()
+        if quorum.for_test(self.test_context) == quorum.zk or interbroker_security_protocol != 'SSL':
+            self.kafka.restart_cluster()
+        else:
+            # Raft-based metadata quorum with SSL communication between quorum and broker
+            # will simply fail to work due to TLS hostname mismatch
+            self.kafka.remote_controller_quorum.restart_cluster()
```

Comment:
As I understand it, this test wants to mess up the inter-broker listener and see that client's get the correct error. However, in raft mode, we can't load any metadata if the inter-broker listener is non-functional since we cannot connect to the raft quorum. This is analogous to breaking the connection to ZK in ZK mode. 

Maybe we should just take the raft mode out of this test and write a new test that exercises similar behavior? Perhaps we could let the inter-broker listener stay properly configured, but mis-configure the controller listener. WDYT?

Reply:
Yeah, I agree that mucking with the broker-controller connectivity is analogous to mucking with the non-Raft brokers' connection to ZooKeeper.  We currently don't check that Kafka will not start if hostname verification fails to a TLS-enabled Zookeeper quorum, but that type of a test -- Raft and Zookeeper -- would fit in a "security system test" as we are doing here.  I think there's room for checking to make sure we can't start with hostname verification failing to the Raft controller quorum.  I don't know if it's worth adding a ZooKeeper case given that TLS to ZooKeeper just became supported in AK 2.7 (meaning adoption is light) and we are moving away from ZooKeeper.  Reaction to these thoughts?

Aside from the above, we could consider leaving Broker-to-Raft-Controller connectivity unperturbed/functional and make hostname verification fail for the inter-broker security protocol as you suggest.  We could add that after we get the ZooKeeper version working again.  I could "fix" that test and add the Raft equivalent, but Boyang needs to take a look to see if the auto topic creation manager changes are causing something to happen that should not.